src/Pos/Wallet/Web/Tracking.hs

```diff
@@ -332,15 +332,15 @@ trackingApplyTxs (getEncInfo -> encInfo) allAddresses getDiff getTs txs =
             tx@(UnsafeTx (NE.toList -> inps) (NE.toList -> outs) _) = taTx
             txId = hash tx
 
-        resolvedInputs <- catMaybes <$> mapM (\tin -> fmap (tin, ) <$> utxoGet tin) inps
+        resolvedInputs <- catMaybes <$> mapM (\tin -> (tin, ) <<$>> utxoGet tin) inps
         let txOutgoings = map txOutAddress outs
-            txInputs = map (toaOut . snd) resolvedInputs
+            txInputs = mergeTxOuts $ map (toaOut . snd) resolvedInputs
```

Comment:
You could simply `ordNub` the `inps` before, and you'll guarantee that no resolved inputs or outputs are the same.